src/config-type.ts

```diff
@@ -48,6 +48,12 @@ export interface CommonConfig<TAG extends string = string> {
    * @see defaultResultHandler
    */
   errorHandler?: AbstractResultHandler;
+  /**
+   * @desc Interprets an error into a status code for HttpError handled by a ResultHandler
+   * @default error instanceof InputValidationError ? 400 : 500
+   * @todo naming
+   * */
+  getStatusCode?: (error: Error) => number;
```

Comment:
MAYBE!
Just maybe.
It may be that we can just remove custom errors and always use HttpError, but allow to configure it for certain cases.
Just a though

Reply:
The interfaces might get easier, no extra processing and wrapping.
Think about it.